src/EFCore/Query/ExpressionVisitors/Internal/ParameterExtractingExpressionVisitor.cs

```diff
@@ -294,6 +294,22 @@ protected override Expression VisitMember(MemberExpression memberExpression)
             return ExtractParameters(queryable.Expression);
         }
 
+        /// <summary>
+        ///     This API supports the Entity Framework Core infrastructure and is not intended to be used
+        ///     directly from your code. This API may change or be removed in future releases.
+        /// </summary>
+        protected override Expression VisitConditional(ConditionalExpression conditionalExpression)
```

Comment:
I thought about this more in connection of the test which changed. This is actually incorrect fix.
All we should do here is to check if conditionalExpression is evaluatable then do tryextractparameter. Similar to how ArrayIndex is working and you are applying fix for coalesce.

The reason is, the test whose sql is changing, by effectively trimming tree of range variable, now we have 2 different trees for different values of same parameter, which causes us to recompile the tree again, effectively defeating the purpose of parameterization.
```suggestion
        protected override Expression VisitConditional(ConditionalExpression conditionalExpression)
        {
            if (_partialEvaluationInfo.IsEvaluatableExpression(conditionalExpression)
            {
                return TryExtractParameter(conditionalExpression);
            }
            
            return base.VisitConditional(conditionalExpression);
        }
```